sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -112,10 +112,14 @@ public void setLevel(final @Nullable SentryLevel level) {
    *
    * @param transaction the transaction
    */
-  public void setTransaction(final @NotNull String transaction) {
+  public void setTransaction(final @Nullable String transaction) {
     final ITransaction tx = this.transaction;
     if (tx != null) {
-      tx.setName(transaction);
+      if (transaction != null) {
+        tx.setName(transaction);
+      } else {
+        this.setTransaction((ITransaction) null);
```

Comment:
I think this could be unwanted side effect.

If I have a Sentry error monitoring instrumentation that sets transaction to null, I add Sentry performance monitoring and I'd be kicking out a transaction bound to the scope without knowing.

Sounds like there are not good solutions because of this odd state where we bind an object adn a string to represent transaction due to backward compatibility but at least we should log a warning here, but potentially not doing this. Just setting the scope transaction to null and not dropping the transaction object

Reply:
Maybe `transaction` name parameter can be set to `@NotNull`? We still give option to explicitly call `Scope#clearTransaction`.

Reply:
Sounds like that's the best solution. @marandaneto ?

Reply:
having `Scope#clearTransaction` makes sense, wondering if it'd exist such an use case that you want to clean the String but not the transaction itself, I don't think so, so not nullable makes sense for me